### DISCUSSION TITLE
AppLambdaFunction is returning an error Role that I don’t understand or how to debug

### DISCUSSION THREAD

**Discussion by diegoug** [serverless/serverless#9534] [Q&A]
Hello, I am following a Vadym Khodak manual where django is deployed on a derverless server but I get an error in a part that I do not understand, I don’t know how to debug and I don’t know how to fix it, please help!

https://blog.vadymkhodak.com/deploy-django-app-on-aws-lambda-using-serverless-part-1

the error is this

AppLambdaFunction
CREATE_FAILED
Properties validation failed for resource AppLambdaFunction with message: #/Role: failed validation constraint for keyword [pattern]

Traceback:
$ serverless deploy -s production
Serverless: Deprecation warning: Support for “package.include” and “package.exclude” will be removed with next major release. Please use “package.patterns” instead
More Info: /framework/docs/deprecations/#NEW_PACKAGE_PATTERNS
Serverless: Deprecation warning: Starting with version 3.0.0, following property will be replaced:
“provider.role” → “provider.iam.role”
More Info: /framework/docs/deprecations/#PROVIDER_IAM_SETTINGS
Serverless: Adding Python requirements helper to …
Serverless: Generated requirements from /opt/app/requirements.txt in /opt/app/.serverless/requirements.txt…
Serverless: Using static cache of requirements found at /root/.cache/serverless-python-requirements/5c3_slspyc …
Serverless: Zipping required Python packages for …
Serverless: Using Python specified in “runtime”: python3.7
Serverless: Packaging Python WSGI handler…
Serverless: Packaging service…
Serverless: Excluding development dependencies…
Serverless: Removing Python requirements helper from …
Serverless: Injecting required Python packages to package…
Serverless: Uploading CloudFormation file to S3…
Serverless: Uploading artifacts…
Serverless: Uploading service app.zip file to S3 (21.41 MB)…
Serverless: Validating template…
Serverless: Creating Stack…
Serverless: Checking Stack create progress…
…
Serverless: Operation failed!
Serverless: View the full error output: /cloudformation/home?region=us-east-1#/stack/detail?stackId=arn%3Aaws%3Acloudformation%3Aus-east-1%3394481%3Astack%2Fdjango-aws-lambda-production%2-beee-11eb-baa0-12

Serverless Error ----------------------------------------

An error occurred: AppLambdaFunction - Properties validation failed for resource AppLambdaFunction with message:
#/Role: failed validation constraint for keyword [pattern].

Get Support --------------------------------------------
Docs: docs.serverless.com
Bugs: /serverless/serverless/issues
Issues:

Your Environment Information ---------------------------
Operating System: linux
Node Version: 14.17.0
Framework Version: 2.43.1
Plugin Version: 5.1.3
SDK Version: 4.2.2
Components Version: 3.10.1

**Comment by pgrzesik** [comment]
Hello @diegoug - this tracker is intended for enhancement proposals and bug reports. I believe this is more suited for a discussion - I'm going to transfer it there.
